test/unit/animation.js

```diff
@@ -332,6 +332,24 @@
     }, 1000);
   });
 
+  QUnit.test('animate with list of values', function(assert) {
+    var done = assert.async();
+
+    fabric.util.animate({
+      startValue: [1, 2, 3],
+      endValue: [2, 4, 6],
+      onValue: [1, 1, 1],
+      duration: 96,
+      onChange: function(currentValue) {
+        assert.equal(currentValue.length, 3)
```

Comment:
@rockerBOO could you add an assertion for `fabric.runningAnimations`
It's in the test file.
You should check that the values have updated accordingly there.

Reply:
And maybe we want to concat the array so that it isn`t the same ref used by animate.
Hmm....

Reply:
The same applies to the callbacks onChange etc.

Reply:
I would add a test:

```js
fabric.util.animate({
  startValue: [1, 2, 3],
  endValue: [2, 4, 6],
  onChange: function([a, b, c]) {
    a=200;
  },
    onComplete([a]) {
     assert.equal(a, 2);
    }
});

```

Reply:
Added a few more tests for checking if fabric.runningAnimations is valid and that the values are valid. The interstitial values would be hard to test but I think this checks for what could be valid values.

Reply:
Whoops missed some of this.

> ```js
> assert.lt(a, 2);
>     a=200;
> ```

What does this mean? I am not sure. Is what I have adequate? I don't think there should be any sharing of arrays as I'm using map to transpose the easing values.  Let me know. Thanks!

Reply:
As you can see here:
https://github.com/fabricjs/fabric.js/blob/7db2515f09c4950a31fe0b1db1dddedf2fcd0240/src/util/animate.js#L174

the array you've created will be set to `runningAnimations`. This isn't good, it needs to be copied. Same goes for values passed to the callbacks. Or else a change in the callback will change inner logic of the animation tick - undesirable!

That's what we need to write a test for both `runningAnimations` and the callbacks - making sure the arrays aren't equal, only `deepEqual`



> Whoops missed some of this.
> 
> > ```js
> > assert.lt(a, 2);
> >     a=200;
> > ```
> 
> What does this mean? I am not sure. Is what I have adequate? I don't think there should be any sharing of arrays as I'm using map to transpose the easing values. Let me know. Thanks!

This bit of code mutates the values passed from the animation driver to the callback.
If all goes well the mutation will live only inside the callback so the next tick will assert that `a` is `lt` (lesser than) 2.
If it's 200 then the arrays are shared.

Reply:
```js
assert.equal(fabric.runningAnimations.length, 1, 'runningAnimations should not be empty');
var context=fabric.runningAnimations[0];
// check that context is deep equal to the values but that the arrays are not the same object

```

Reply:
Ok I think I got it all working now. Thanks!